modules/monitoring-supported-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,314 @@
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: sigv4-credentials
+  namespace: openshift-monitoring
+stringData:
+  access: <AWS_access_key> <1>
+  secret: <AWS_secret_key> <2>
+type: Opaque
+----
+<1> The AWS API access key.
+<2> The AWS API secret key.
+
+The following shows sample AWS Signature Version 4 remote write authentication settings that use a `Secret` object named `sigv4-credentials` in the `openshift-monitoring` namespace:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://authorization.example.com/api/write"
+        sigv4:
+          region: <AWS_region> <1>
+          accessKey:
+            name: sigv4-credentials <2>
+            key: access <3>
+          secretKey:
+            name: sigv4-credentials <2>
+            key: secret <3>
+          profile: <AWS_profile_name> <4>
+          roleArn: <AWS_role_arn> <5>
+----
+<1> The AWS region.
+<2> The name of the `Secret` object containing the AWS API access credentials.
+<3> The AWS API access credentials contained in the specified `Secret` object.
+<4> The name of the AWS profile that is being used to authenticate.
+<5> The unique identifier for the Amazon Resource Name (ARN) assigned to your role.
+
+
+.Sample YAML for basic authentication
+
+The following shows sample basic authentication settings for a `Secret` object named `rw-basic-auth` in the `openshift-monitoring` namespace: 
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rw-basic-auth
+  namespace: openshift-monitoring
+stringData:
+  user: <basic_username> <1>
+  password: <basic_password> <2>
+type: Opaque
+----
+<1> The username.
+<2> The password.
+
+The following sample shows a `basicAuth` remote write configuration that uses a `Secret` object named `rw-basic-auth` in the `openshift-monitoring` namespace.
+It assumes that you have already set up authentication credentials for the endpoint. 
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://basicauth.example.com/api/write"
+        basicAuth:
+          username:
+            name: rw-basic-auth <1>
+            key: user <2>
+          password:
+            name: rw-basic-auth <1>
+            key: password <3>
+----
+<1> The name of the `Secret` object that contains the authentication credentials.
+<2> The username contained in the specified `Secret` object.
+<3> The password contained in the specified `Secret` object.
+
+.Sample YAML for bearer token authentication using a `Secret` Object
+
+The following shows bearer token settings for a `Secret` object named `rw-bearer-auth` in the `openshift-monitoring` namespace:
```

Comment:
I think that it's fine to mention bearer token from this point (to keep the example concrete).

Reply:
@simonpasquier Would it be more accurate to say `bearer token authorization` rather than `bearer token authentication`?

Reply:
The configuration entry is called `authorization` because this is the name of the HTTP header but this is misleading since the mechanism is really meant to authenticate the HTTP client (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization). So while it's `authorization` in the configuration file, I believe that the documentation should use authentication/authenticate.